diesel_cli/src/print_schema.rs

```diff
@@ -326,16 +368,57 @@ impl Display for CustomTypeList {
             }
             #[cfg(feature = "mysql")]
             Backend::Mysql => {
-                let _ = (&f, self.with_docs);
-                for t in &self.types_sorted {
-                    eprintln!("Encountered unknown type for Mysql: {}", t.sql_name);
+                if self.with_docs {
+                    writeln!(f, "/// A module containing custom SQL type definitions")?;
+                    writeln!(f, "///")?;
+                    writeln!(f, "/// (Automatically generated by Diesel.)")?;
                 }
-                unreachable!(
-                    "Mysql only supports a closed set of types.
-                         If you ever see this error message please open an \
-                         issue at https://github.com/diesel-rs/diesel containing \
-                         a dump of your schema definition."
-                )
+
+                let mut out = PadAdapter::new(f);
+                writeln!(out, "pub mod sql_types {{")?;
+
+                for (idx, ct) in self.types_sorted.iter().enumerate() {
+                    let t = &ct.column_type;
+
+                    if idx != 0 {
+                        writeln!(out)?;
+                    }
+
+                    if self.with_docs {
+                        if let Some(ref schema) = t.schema {
+                            writeln!(out, "/// The `{}.{}` SQL type", schema, t.sql_name)?;
+                        } else {
+                            writeln!(out, "/// The `{}` SQL type", t.sql_name)?;
+                        }
+                        writeln!(out, "///")?;
+                        writeln!(out, "/// (Automatically generated by Diesel.)")?;
+                    }
+
+                    writeln!(out, "#[derive({})]", self.derives.join(", "))?;
+
+                    let mysql_name = {
+                        let mut c = t.sql_name.chars();
+
+                        match c.next() {
+                            None => String::new(),
+                            Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
+                        }
+                    };
+
+                    if let Some(ref schema) = t.schema {
+                        writeln!(
+                            out,
+                            "#[diesel(mysql_type(name = \"{mysql_name}\", schema = \"{schema}\"))]"
+                        )?;
```

Comment:
This can not happen as we do not support schemas for mysql. Also the derive does not support that parameter. (That's mostly a postgres thing).